catalog/transformer/catalog-transformer-zip/src/main/java/org/codice/ddf/catalog/transformer/zip/ZipCompression.java

```diff
@@ -64,12 +68,36 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(ZipCompression.class);
 
+  private static final String TRANSFORMER_ID = "transformerId";
+
   private CatalogFramework catalogFramework;
 
   private JarSigner jarSigner;
 
-  public ZipCompression(JarSigner jarSigner) {
+  private List<ServiceReference> metacardTransformers;
+
+  private BundleContext bundleContext;
+
+  private static MimeType mimeType;
+
+  private static final File DEFAULT_PATH =
+      Paths.get(System.getProperty("ddf.home"), "data", "tmp", "export.zip").toFile();
```

Comment:
I don't know that we need to continue to write out a file unless there is no good way to manage the zip in memory.  There is a size concern for sure but the result of this transformer reads the entire zip data into memory anyway to return it.  That said, if we keep it writing to a file and a specific file is not specified via the arguments then this default file would be referenced by multiple requests.  That would cause multiple export requests to smash this same file and would end up failing and/or be corrupt.  At a minimum, this would have to get a random UUID like file name to avoid that situation.  Additionally, we will want to clean it up after the transform is done so that it doesn't leave a bunch of random zip files on the filesystem.